server/src/main/java/io/crate/replication/logical/action/PublicationsStateAction.java

```diff
@@ -163,13 +175,41 @@ protected ClusterBlockException checkBlock(Request request,
         }
 
         @VisibleForTesting
-        static List<RelationName> resolveRelationsNames(Publication publication, Schemas schemas) {
+        static User ensurePublicationOwnerExists(String publicationOwner, UserLookup userLookup, String publicationName) {
+            var user = userLookup.findUser(publicationOwner);
+            if (user == null) {
+                throw new IllegalStateException(
+                    String.format(Locale.ENGLISH, "User %s owning the publication %s is not found, stopping replication.",
+                        publicationOwner,
+                        publicationName
+                    ));
+            }
+            return user;
+        }
+
+        @VisibleForTesting
+        static User ensureSubscriberExists(String subscriber, UserLookup userLookup, String publicationName) {
+            var user = userLookup.findUser(subscriber);
+            if (user == null) {
+                throw new IllegalStateException(
+                    String.format(Locale.ENGLISH, "User %s subscribed to the publication %s is not found, stopping replication.",
+                        subscriber,
+                        publicationName
+                    ));
+            }
+            return user;
+        }
+
+        @VisibleForTesting
+        static List<RelationName> resolveRelationsNames(Publication publication, Schemas schemas, User publicationOwner, User subscriber) {
+
             List<RelationName> relationNames;
             if (publication.isForAllTables()) {
                 relationNames = InformationSchemaIterables.tablesStream(schemas)
                     .filter(t -> {
                         if (t instanceof DocTableInfo dt) {
                             boolean softDeletes;
+                            // All checks are aligned for both cases (ALL TABLES and fixed) except soft deletes check.
```

Comment:
Looks like soft deletes check is not done fixed tables - should be added?

Reply:
This is already done at the analyzer, see https://github.com/crate/crate/blob/logical-replication/server/src/main/java/io/crate/replication/logical/analyze/LogicalReplicationAnalyzer.java#L81.
Please remove the comment or change it to state that for concrete tables, this can be and is already done at the analyzer.